model/patch_lifecycle.go

```diff
@@ -92,14 +92,28 @@ func ValidateTVPairs(p *Project, in []TVPair) error {
 // do not exist yet out of the set of pairs. No tasks are added for builds which already exist
 // (see AddNewTasksForPatch).
 func AddNewBuildsForPatch(ctx context.Context, p *patch.Patch, patchVersion *Version, project *Project, tasks TaskVariantPairs) error {
-	_, _, err := addNewBuilds(ctx, batchTimeTasksAndVariants{}, patchVersion, project, tasks, p.SyncAtEndOpts, "")
+	projectRef, err := FindOneProjectRef(project.Identifier)
```

Comment:
Assuming that the tasks and build will be part of the same project, Is there a way to extract out the FindOneProjectRef logic, or would that require too much refactoring?